File: Data Structures & Algorithms/minimum-stack/submission-0.cpp
class MinStack {
private:
    stack<int> _mainStack;
    stack<int> _minStack;

public:
    MinStack() {}
    
    void push(int val) {
        _mainStack.push(val);
        int min = INT_MAX;
        if (!_minStack.empty()) {
            min = _minStack.top();
        }

        if (val < min) {
            min = val;
        }

        _minStack.push(min);
    }
    
    void pop() {
        _mainStack.pop();
        _minStack.pop();
    }
    
    int top() {
        return _mainStack.top();
    }
    
    int getMin() {
        return _minStack.top();
    }
};

File: Data Structures & Algorithms/validate-parentheses/submission-0.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        par = {')':'(', '}':'{', ']':'['}

        for c in s:
            if len(stack) > 0 and par.get(c, None) == stack[len(stack) - 1]:
                stack.pop()
            else:
                stack.append(c)
        
        return len(stack) == 0